Make.inc

```diff
@@ -375,9 +375,21 @@ LD := link
 endif #USEMSVC
 RANLIB := $(CROSS_COMPILE)ranlib
 
+# file extensions
+ifeq ($(OS), WINNT)
+  SHLIB_EXT = dll
+  SHELL_EXT = bat
+else ifeq ($(OS), Darwin)
+  SHLIB_EXT = dylib
+  SHELL_EXT = sh
+else
+  SHLIB_EXT = so
+  SHELL_EXT = sh
+endif
+
 
 # if not absolute, then relative to the directory of the julia executable
-JCPPFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys.ji\""
+JCPPFLAGS += "-DJL_SYSTEM_IMAGE_PATH=\"$(build_private_libdir_rel)/sys.$(SHLIB_EXT)\""
```

Comment:
Yes, thanks. Actually I wonder if there should be some flexibility in the default system image name, trying both .ji and .dll so it can load whichever exists.